src/shared/Services/StoreCacheService.cs

```diff
@@ -0,0 +1,62 @@
+using chia.dotnet;
+
+namespace dig;
+
+public class StoreCacheService(DataLayerProxy dataLayer,
+                                FileCacheService fileCache,
+                                ILogger<CacheService> logger)
+{
+    private readonly DataLayerProxy _dataLayer = dataLayer;
+    private readonly FileCacheService _fileCache = fileCache;
+    private readonly ILogger<CacheService> _logger = logger;
+
+    public async Task VerifyStoreCache(string storeId, CancellationToken cancellationToken)
+    {
+        if (!await StoreCacheIsValid(storeId, cancellationToken))
+        {
+            _logger.LogWarning("Invalidating cache for {StoreId}", storeId.SanitizeForLog());
+            InvalidateStoreCache(storeId);
+        }
+    }
+
+    public async Task<bool> StoreCacheIsValid(string storeId, CancellationToken cancellationToken)
+    {
+        var lastRoot = await _fileCache.GetValueAsync<RootHash>($"{storeId}-last-root", cancellationToken);
+        var currentRoot = await _dataLayer.GetRoot(storeId, cancellationToken);
```

Comment:
i know we discussed not caching the root, but im almost wondering if we should cache it like 15 seconds or something.
That way burst requests wont hit this rpcs to fast.

(anytime your loading a webpage from datalayer, its going to create a burst of requests all at once while your loading the page)